aroma/aroma.py

```diff
@@ -187,58 +141,25 @@ def aroma_workflow(
             "-------------- ICA-AROMA IS CANCELED ------------\n"
         )
 
-    # Define/create mask. Either by making a copy of the specified mask, or by
-    # creating a new one.
-    new_mask = op.join(out_dir, "mask.nii.gz")
-    if mask:
-        shutil.copyfile(mask, new_mask)
-    elif in_feat and op.isfile(op.join(in_feat, "example_func.nii.gz")):
-        # If a Feat directory is specified, and an example_func is present use
-        # example_func to create a mask
-        bet_command = "{0} {1} {2} -f 0.3 -n -m -R".format(
-            op.join(fsl_dir, "bet"),
-            op.join(in_feat, "example_func.nii.gz"),
-            op.join(out_dir, "bet"),
-        )
-        os.system(bet_command)
-        os.rename(op.join(out_dir, "bet_mask.nii.gz"), new_mask)
-        if op.isfile(op.join(out_dir, "bet.nii.gz")):
-            os.remove(op.join(out_dir, "bet.nii.gz"))
-    else:
-        if in_feat:
-            LGR.warning(
-                " - No example_func was found in the Feat directory. "
-                "A mask will be created including all voxels with varying "
-                "intensity over time in the fMRI data. Please check!\n"
-            )
-        math_command = "{0} {1} -Tstd -bin {2}".format(
-            op.join(fsl_dir, "fslmaths"), in_file, new_mask
-        )
-        os.system(math_command)
-
-    # Run ICA-AROMA
-    LGR.info("Step 1) MELODIC")
-    utils.runICA(fsl_dir, in_file, out_dir, mel_dir, new_mask, dim, TR)
-
-    LGR.info("Step 2) Automatic classification of the components")
-    LGR.info("  - registering the spatial maps to MNI")
-    mel_IC = op.join(out_dir, "melodic_IC_thr.nii.gz")
-    mel_IC_MNI = op.join(out_dir, "melodic_IC_thr_MNI2mm.nii.gz")
-    utils.register2MNI(fsl_dir, mel_IC, mel_IC_MNI, affmat, warp)
+    # Load more inputs
+    motion_params = utils.load_motpars(mc, source=mc_source)  # T x 6
+    mixing = np.loadtxt(mixing)  # T x C
+    component_maps = nib.load(component_maps)  # X x Y x Z x C
+    assert mixing.shape[1] == component_maps.shape[3], f"{mixing.shape}, {component_maps.shape}"
```

Comment:
These assertions within the executed code (i.e., not a test) should be better raised as python errors. The reason is that an error is much more informative of what happened (e.g., a `ValueError` in this particular line because the mixing and component dimensions do not match, or a `FileNotFoundError` are way more clear than assertion errors and a description that might be more or less informative)

Reply:
That's fair. I replaced the assertions outside of tests in 9633a32.